sbin/pfmon

```diff
@@ -408,18 +409,24 @@ sub _runtask {
             pause;
             next;
         }
+        my $final_interval = $time_taken > 0 ? $interval - int($time_taken+1) : $interval;
+        if($final_interval < 0) {
+            $final_interval = 0;
+        }
         # run these functions every $maintenance_interval
-        $logger->trace("$id is sleeping for $interval");
-        alarm $interval;
+        $logger->trace("$id is sleeping for $final_interval ($interval from configuration - $time_taken)");
+        alarm $final_interval;
```

Comment:
More details.
If the alarm value is 0
Then no alarm is scheduled.
This will cause pause to stay paused until the process gets a signal.